[DOC]: updated tutorial notebooks

aggreen opened the pull request:
I updated the tutorial python notebooks to reflect developments in the pipeline and to be more accessible to outside users. I wanted to give you both the option to review before I pull these into develop. If you'd rather not review, that's fine too. 

The main changes are:
 
- Added documentation for the folding and mutate stages
- Added documentation for the complexes pipeline
- Added instructions for setting up tools, databses, and environment section of the config file to the running_jobs.ipynb notebook
- Added an explanation of pipeline stages and restarting runs from various stages to the running_jobs.ipynb notebook
- Expanded wording and parameter explanation in the running_jobs.ipynb notebook to be friendlier to outside users

I'm talking to @kpgbrock about getting an explanation of the most important files for folding. That will be added to output_files_tutorial.ipynb in the next couple of days.

thomashopf commented:
Cool, I will have a look this weekend!

Am 05.03.2018 20:32 schrieb "Anna Green" <notifications@github.com>:

> I updated the tutorial python notebooks to reflect developments in the
> pipeline and to be more accessible to outside users. I wanted to give you
> both the option to review before I pull these into develop. If you'd rather
> not review, that's fine too.
>
> The main changes are:
>
>    - Added documentation for the folding and mutate stages
>    - Added documentation for the complexes pipeline
>    - Added instructions for setting up tools, databses, and environment
>    section of the config file to the running_jobs.ipynb notebook
>    - Added an explanation of pipeline stages and restarting runs from
>    various stages to the running_jobs.ipynb notebook
>    - Expanded wording and parameter explanation in the running_jobs.ipynb
>    notebook to be friendlier to outside users
>
> I'm talking to @kpgbrock <https://github.com/kpgbrock> about getting an
> explanation of the most important files for folding. That will be added to
> output_files_tutorial.ipynb in the next couple of days.
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/debbiemarkslab/EVcouplings/pull/143
> Commit Summary
>
>    - [DOC]: updated tutorial notebooks
>    - [DOC]: ambiguous protocol names in align_1 and align_2 stage of
>    complexes example config fixed.
>
> File Changes
>
>    - *M* config/sample_config_complex.txt
>    <https://github.com/debbiemarkslab/EVcouplings/pull/143/files#diff-0>
>    (6)
>    - *M* notebooks/output_files_tutorial.ipynb
>    <https://github.com/debbiemarkslab/EVcouplings/pull/143/files#diff-1>
>    (145)
>    - *M* notebooks/running_jobs.ipynb
>    <https://github.com/debbiemarkslab/EVcouplings/pull/143/files#diff-2>
>    (203)
>    - *M* notebooks/running_jobs_complexes.ipynb
>    <https://github.com/debbiemarkslab/EVcouplings/pull/143/files#diff-3>
>    (157)
>
> Patch Links:
>
>    - https://github.com/debbiemarkslab/EVcouplings/pull/143.patch
>    - https://github.com/debbiemarkslab/EVcouplings/pull/143.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/debbiemarkslab/EVcouplings/pull/143>, or mute the
> thread
> <https://github.com/notifications/unsubscribe-auth/AHFvctMCsXVUze1yeedUtXHQ5z2p1Zprks5tbZKvgaJpZM4SdGrC>
> .
>

thomashopf commented on notebooks/running_jobs.ipynb:
```diff
@@ -17,41 +31,65 @@
-    "### Parameters that might need to be modified (non-exhaustive):\n",
+    "### Setup\n",
+    "\n",
+    "Before you run any jobs, you must set up baths to external tools and databases, and specify your batch submission engine (if applicable). Modify the following sections of your config file:\n",
```
paths, not baths ;)

thomashopf commented on notebooks/running_jobs.ipynb:
```diff
@@ -17,41 +31,65 @@
+    "* __minimum_column_coverage__: Only include alignment columns with at least x% residues (rather than gaps) during model inference. (formerly called \"-m\" in buildali, but now the other way round: -m 30 corresponds to minimum_column_coverage=70)\n",
+    "\n",
+    "2) in \"couplings\" section:\n",
+    "* __theta__: Clustering threshold for downweighting redudant sequences (Meff computation). For example, 0.8 will cluster sequencesat a 80% sequence identity cutoff (note that this is 1 - theta used in previous pipelines)\n",
```
space between sequences and at

thomashopf commented on notebooks/running_jobs.ipynb:
```diff
@@ -17,41 +31,65 @@
+    "3) in \"compare\" section:\n",
+    "* __by_alignment__: If True, structures for comparison will be identified by homology search; otherwise use only structures for given sequence_id (must be UniProt ID or AC)\n",
+    "* __pdb_alignment_method__: jackhmmer or hmmsearch. Method for searching the PDB sequence database for homologous structures. Jackhmmer is more stringent, hmmsearch is more lenient. \n",
+    "* __pdb_ids__: Used if by_alignment is False. If pdb_ids is None, compares to all PDB structures for given sequence_id. If list of PDB IDs, compare to that subset of structures only.\n",
```
Actually this will be used no matter if alignment or lookup by Uniprot ID/AC is used - it just acts as a filter on the structure list independent of how it was made

thomashopf commented on notebooks/running_jobs.ipynb:
```diff
@@ -89,11 +127,22 @@
+   "metadata": {},
+   "outputs": [
+    {
+     "ename": "FileNotFoundError",
```
looks like this blows up because sample_config.txt is now sample_config_monomer.txt

thomashopf commented on notebooks/running_jobs.ipynb:
```diff
@@ -169,14 +223,77 @@
+    "\n",
+    "__compare__: Searches for structures against which to compare the couplings, visualizes couplings in a contact map, calculates precision of couplings against structure. \n",
+    "\n",
+    "__mutate__: Generates and visualizes matrices of the predicted effects of mutations at every site. \n",
```
would also add that you can predict a user-supplied list of mutations (any order) here too

thomashopf commented on notebooks/running_jobs.ipynb:
```diff
@@ -169,14 +223,77 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "A c"
```
who am I? ;)

aggreen commented:
Thanks. I have addressed these changes in my most recent commit, and added
documentation of the output of the folding stage from @kpgbrock

On Fri, Mar 9, 2018 at 6:16 AM, thomashopf <notifications@github.com> wrote:

> *@thomashopf* approved this pull request.
>
> Very nice! Annotated a couple of small things (typos, one missing path,
> one parameter correction) but otherwise these enhancements should be really
> helpful to new users
> ------------------------------
>
> In notebooks/running_jobs.ipynb
> <https://github.com/debbiemarkslab/EVcouplings/pull/143#discussion_r173422194>
> :
>
> >      "\n",
>      "> For an example configuration file, see config/sample_config.txt in this repository.\n",
>      "\n",
> -    "### Parameters that might need to be modified (non-exhaustive):\n",
> +    "### Setup\n",
> +    "\n",
> +    "Before you run any jobs, you must set up baths to external tools and databases, and specify your batch submission engine (if applicable). Modify the following sections of your config file:\n",
>
> paths, not baths ;)
> ------------------------------
>
> In notebooks/running_jobs.ipynb
> <https://github.com/debbiemarkslab/EVcouplings/pull/143#discussion_r173422634>
> :
>
> > +    "### Additional parameters to modify (non-exhaustive):\n",
> +    "\n",
> +    "The following additional parameters are the most important ones to consider modifying when running your job. See example configuration files for full parameter explanations:\n",
> +    "\n",
> +    "1) In \"alignment\" section:\n",
> +    "* __domain_threshold__ (bitscore or evalue): Inclusion threshold for sequence alignment. 0.5 is a good starting point for bitscore, e.g. use 0.3 for more inclusive and 0.7 or 0.8 for less inclusive alignment\n",
> +    "* __sequence_threshold__ (bitscore or evalue): typically should be same as domain_threshold\n",
> +    "* __use_bitscores__: default true, set to false to use evalues for domain_threshold and sequence_threshold\n",
> +    "* __iterations__: number of jackhmmer iterations\n",
> +    "* __database__: uniref, uniprot (see databases section for options or to modify path)\n",
> +    "* __seqid_filter__: Filter alignment at x% sequence identity. Leave empty or set to 100 not to filter any sequences.\n",
> +    "* __minimum_sequence_coverage__: Only keep sequences that align to at least x% of the target sequence (formerly called \"-f\" in buildali)\n",
> +    "* __minimum_column_coverage__: Only include alignment columns with at least x% residues (rather than gaps) during model inference. (formerly called \"-m\" in buildali, but now the other way round: -m 30 corresponds to minimum_column_coverage=70)\n",
> +    "\n",
> +    "2) in \"couplings\" section:\n",
> +    "* __theta__: Clustering threshold for downweighting redudant sequences (Meff computation). For example, 0.8 will cluster sequencesat a 80% sequence identity cutoff (note that this is 1 - theta used in previous pipelines)\n",
>
> space between sequences and at
> ------------------------------
>
> In notebooks/running_jobs.ipynb
> <https://github.com/debbiemarkslab/EVcouplings/pull/143#discussion_r173423137>
> :
>
> > +    "* __iterations__: number of jackhmmer iterations\n",
> +    "* __database__: uniref, uniprot (see databases section for options or to modify path)\n",
> +    "* __seqid_filter__: Filter alignment at x% sequence identity. Leave empty or set to 100 not to filter any sequences.\n",
> +    "* __minimum_sequence_coverage__: Only keep sequences that align to at least x% of the target sequence (formerly called \"-f\" in buildali)\n",
> +    "* __minimum_column_coverage__: Only include alignment columns with at least x% residues (rather than gaps) during model inference. (formerly called \"-m\" in buildali, but now the other way round: -m 30 corresponds to minimum_column_coverage=70)\n",
> +    "\n",
> +    "2) in \"couplings\" section:\n",
> +    "* __theta__: Clustering threshold for downweighting redudant sequences (Meff computation). For example, 0.8 will cluster sequencesat a 80% sequence identity cutoff (note that this is 1 - theta used in previous pipelines)\n",
> +    "* __iterations__: how many plmc iterations to run\n",
> +    "* __ignore_gaps__: exclude gaps from EC calculation\n",
> +    "* __save_model__: save binary file with model parameters or not. These files can be very large.\n",
> +    "\n",
> +    "3) in \"compare\" section:\n",
> +    "* __by_alignment__: If True, structures for comparison will be identified by homology search; otherwise use only structures for given sequence_id (must be UniProt ID or AC)\n",
> +    "* __pdb_alignment_method__: jackhmmer or hmmsearch. Method for searching the PDB sequence database for homologous structures. Jackhmmer is more stringent, hmmsearch is more lenient. \n",
> +    "* __pdb_ids__: Used if by_alignment is False. If pdb_ids is None, compares to all PDB structures for given sequence_id. If list of PDB IDs, compare to that subset of structures only.\n",
>
> Actually this will be used no matter if alignment or lookup by Uniprot
> ID/AC is used - it just acts as a filter on the structure list independent
> of how it was made
> ------------------------------
>
> In notebooks/running_jobs.ipynb
> <https://github.com/debbiemarkslab/EVcouplings/pull/143#discussion_r173423406>
> :
>
> > @@ -89,11 +127,22 @@
>    },
>    {
>     "cell_type": "code",
> -   "execution_count": null,
> -   "metadata": {
> -    "collapsed": false
> -   },
> -   "outputs": [],
> +   "execution_count": 4,
> +   "metadata": {},
> +   "outputs": [
> +    {
> +     "ename": "FileNotFoundError",
>
> looks like this blows up because sample_config.txt is now
> sample_config_monomer.txt
> ------------------------------
>
> In notebooks/running_jobs.ipynb
> <https://github.com/debbiemarkslab/EVcouplings/pull/143#discussion_r173423633>
> :
>
> > +  },
> +  {
> +   "cell_type": "markdown",
> +   "metadata": {},
> +   "source": [
> +    "# Pipeline Stages\n",
> +    "\n",
> +    "The evcouplings pipeline is split into different stages for different aspects of computation. This allows users to run only the desired stages, and to re-start computation from an arbitrary stage. The stages are as follows:\n",
> +    "\n",
> +    "__align__: Creates sequence alignments or reads in an existing alignment, and pre-processes alignment for couplings calculation.\n",
> +    "\n",
> +    "__couplings__: Calculates evolutionary couplings, post-processes output into correct numbering, and fits statistical scoring model to couplings. \n",
> +    "\n",
> +    "__compare__: Searches for structures against which to compare the couplings, visualizes couplings in a contact map, calculates precision of couplings against structure. \n",
> +    "\n",
> +    "__mutate__: Generates and visualizes matrices of the predicted effects of mutations at every site. \n",
>
> would also add that you can predict a user-supplied list of mutations (any
> order) here too
> ------------------------------
>
> In notebooks/running_jobs.ipynb
> <https://github.com/debbiemarkslab/EVcouplings/pull/143#discussion_r173423876>
> :
>
> > +    "Example 2: if you have run \"align\", \"couplings\", and \"compare\", and wish to re-run \"compare\" with different parameters, you can copy your configuration file and then comment out \"align\" and \"couplings\" under \"stages\". You can then modify parameters in the \"compare\" stage and re-submit your job using the newly modified config file. This is somewhat risky, because it will overwrite your previous results for the \"compare\" stage. \n",
> +    "\n",
> +    "By default the evcouplings app will not allow you to submit jobs that overwrite previous results. Use the \"yolo\" flag to live dangerously and force an override.  \n",
> +    "\n",
> +    "```\n",
> +    "evcouplings --yolo [options] <config_file>\n",
> +    "```"
> +   ]
> +  },
> +  {
> +   "cell_type": "code",
> +   "execution_count": null,
> +   "metadata": {},
> +   "outputs": [],
> +   "source": [
> +    "A c"
>
> who am I? ;)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/debbiemarkslab/EVcouplings/pull/143#pullrequestreview-102611825>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHimXvK0PuqpdTHIXWB79JzfykVD6vVjks5tcmSbgaJpZM4SdGrC>
> .
>

thomashopf commented on notebooks/output_files_tutorial.ipynb:
```diff
@@ -1088,6 +1089,81 @@
+    "These files each contain one folded model. File names are formatted as {prefix}\\_{N ECs used for folding}\\_{model number}\\_hMIN.pdb\n",
+    "\n",
+    "### ECs with Secondary Structure Clashes (CouplingScores_with_clashes.csv):\n",
+    "Includes evolutionary couplings sorted by score, with a flag for potential secondary structure clashes (e.g. a coupling between two residues that are in the same helix that cannot be in contact directly due to the helix).  Columns are the same as \\CouplingScores.csv but with an additional column:\n",
```
backslash in front of of CouplingScores.csv

thomashopf commented on notebooks/output_files_tutorial.ipynb:
```diff
@@ -1088,6 +1089,81 @@
+    "- in_model: a boolean flag that displays whether the residue has sufficient coverage in the \n",
+    "\talignment (and is therefore eligible to have evolutionary couplings identified)\n",
+    "\n",
+    "### Structure ranking file (ranking.csv):\n",
```
add short sentence here that larger clusters usually tend to correspond to more reliable predictions (even if obvious)?

thomashopf commented:
Content of folding stage is great, latest commit introduced stray files though: test case, and test_config.txt which is created as output by one of the notebooks

aggreen commented:
changes made

On Fri, Mar 9, 2018 at 10:33 AM, thomashopf <notifications@github.com>
wrote:

> Content of folding stage is great, latest commit introduced stray files
> though: test case, and test_config.txt which is created as output by one of
> the notebooks
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/debbiemarkslab/EVcouplings/pull/143#issuecomment-371846566>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHimXj4xLUft0_5V2gAiTpgZKsgCBABEks5tcqDlgaJpZM4SdGrC>
> .
>
